### ISSUE TITLE
Fade is broken

### ISSUE THREAD

**Issue by kamranayub** [excaliburjs/Excalibur#512]
## Context

The constructor of the `fade` action was determining what direction to go when updating opacity--however, actions are not instantiated at the time they are executed, they are instantiated immediately before any action is executed, so both `fade` actions had the same multiplier.

## Reproduction

```
actor.addDrawing(foo);
actor.opacity = 0;
actor.fade(1, 200).delay(2000).fade(0, 200);
```

## Expected Result

Actor fades in, then fades back out after 2s.

## Actual Result

Actor never fades out.

**Comment by eonarheim** [comment]
This may be fixed as of 0.7, needs confirmation and a test

**Comment by kamranayub** [comment]
Yes, it's definitely broken:

![image](https://cloud.githubusercontent.com/assets/563819/18090338/d9d18ba2-6e89-11e6-9254-243d0ae431ba.png)
